src/java.desktop/share/classes/javax/swing/plaf/basic/BasicButtonListener.java

```diff
@@ -322,7 +322,7 @@ public void actionPerformed(ActionEvent e) {
                 model.setArmed(true);
                 model.setPressed(true);
                 if(!b.hasFocus()) {
-                    b.requestFocus();
+                    b.requestFocus(FocusEvent.Cause.MOUSE_EVENT);
```

Comment:
Can this action be performed using a keyboard or this can be reached only using mouse press?
If this actionPerformed can be reached using keyboard, i think we should add appropriate checks before throwing the focus event cause.

Reply:
I have removed this contentious change as it probably can be used using keyboard too.  For now, just kept the other mouse only focus event cause in this file..